src/servers/src/prom_row_builder.rs

```diff
@@ -0,0 +1,249 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::string::ToString;
+
+use api::prom_store::remote::Sample;
+use api::v1::value::ValueData;
+use api::v1::{
+    ColumnDataType, ColumnSchema, Row, RowInsertRequest, RowInsertRequests, Rows, SemanticType,
+    Value,
+};
+use common_query::prelude::{GREPTIME_TIMESTAMP, GREPTIME_VALUE};
+
+use crate::proto::PromLabel;
+use crate::repeated_field::Clear;
+
+#[derive(Default)]
```

Comment:
Let's add some comments for these structs and fields.

Reply:
fixed https://github.com/GreptimeTeam/greptimedb/pull/3425/commits/571926fd0e2cda3f14633107fa722bf6e73c55c8